### ISSUE TITLE
No support for regional subnet

### ISSUE THREAD

**Issue by cosmindev** [oracle-terraform-modules/terraform-oci-compute-instance#23]
**Issue:**
This module does not work for regional subnets - as it tries to read the AD for the instance from the provided subnet.

**Error:**
When trying to use regional subnets you'll get the following error:
"* module.instance.oci_core_instance.this: Resource 'data.oci_core_subnet.this' does not have attribute 'availability_domain' for variable 'data.oci_core_subnet.this.*.availability_domain'"

**Location:**
The issue is generated at line 16:

```
availability_domain = "${data.oci_core_subnet.this.*.availability_domain[count.index % length(data.oci_core_subnet.this.*.availability_domain)]}"

in the file: /terraform-oci-compute-instance/main.tf
```

**Suggested fix:**

- Add availability_domain as a input parameter
- Use this parameter to set the AD for the instance

**Comment by lucianf** [comment]
+1, the module is unusable with regional subnets (which are the recommended practice) in its current state.

**Comment by kral2** [comment]
fixed in v2.0.4
